File: problems/Valid Parentheses/isValid.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> store = new Stack<Character>();

        for (char letter : s.toCharArray()) {
            if (store.isEmpty() || '(' == letter || '[' == letter || '{' == letter) {
                store.push(letter);
            } else {
                if (')' == letter && '(' != store.pop()) {
                    return false;
                }
                if (']' == letter && '[' != store.pop()) {
                    return false;
                }
                if ('}' == letter && '{' != store.pop()) {
                    return false;
                }
            }
        }

        return store.isEmpty();
    }
}

File: problems/Valid Parentheses/isValid.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> store = new Stack<char>();

        foreach (char letter in s) {
            if (0 == store.Count || 40 == letter || 91 == letter || 123 == letter) {
                store.Push(letter);
            } else {
                char prevLetter = store.Pop();

                if (1 != (letter - prevLetter) && 2 != (letter - prevLetter)) {
                    return false;
                }
            }
        }

        return 0 == store.Count;
    }
}
